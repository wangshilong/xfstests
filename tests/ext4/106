#! /bin/bash
# FS QA Test No. 106
#
# copy from xfs 106, maybe need test more?
#
#-----------------------------------------------------------------------
# Copyright (c) 2005 Silicon Graphics, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "rm -f $tmp.*; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/quota

_supported_fs ext4
_supported_os Linux #IRIX
_require_scratch

rm -f $seqres.full

# real QA test starts here
_scratch_mkfs_ext4 -O quota,project >> $seqres.full 2>&1

# setup a default run
if [ -z "$MOUNT_OPTIONS" ]; then
	export MOUNT_OPTIONS="-o sync"
else
	export MOUNT_OPTIONS="$MOUNT_OPTIONS -o sync"
fi

_scratch_mount >> $seqres.full 2>&1

prid=253
rm -f $SCRATCH_MNT/resv
$XFS_IO_PROG -fc "resvsp 0 200m" $SCRATCH_MNT/resv
src/setproject -v $prid $SCRATCH_MNT/resv

filter_ext4_quota()
{
	grep $SCRATCH_DEV 2>&1 | tr -s '[:space:]' | _filter_scratch
}

test_ext4_quota()
{
	setquota -P $id 100M 100M 0 2 $SCRATCH_MNT
	quota -v -P $id
	setquota -P $id -T $((60*60*24*3)) 0 $SCRATCH_MNT >> $seqres.full 2>&1
	setquota -P $id 300M 300M 0 2 $SCRATCH_MNT
	quota -v -P $id
}

export MOUNT_OPTIONS="sync"
type=p
id=$prid
test_ext4_quota | filter_ext4_quota

umount $SCRATCH_DEV
status=0
exit
