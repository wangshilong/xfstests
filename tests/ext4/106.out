QA output created by 106
 SCRATCH_DEV 204800* 102400 102400 7days 1 0 2 
 SCRATCH_DEV 204800 307200 307200 1 0 2 
